cms/djangoapps/contentstore/views/transcripts_ajax.py

```diff
@@ -154,31 +157,45 @@ def download_transcripts(request):
         log.debug("Can't find item by locator.")
         raise Http404
 
-    subs_id = request.GET.get('subs_id')
-    if not subs_id:
-        log.debug('GET data without "subs_id" property.')
-        raise Http404
-
     if item.category != 'video':
         log.debug('transcripts are supported only for video" modules.')
         raise Http404
 
-    filename = 'subs_{0}.srt.sjson'.format(subs_id)
-    content_location = StaticContent.compute_location(item.location.course_key, filename)
     try:
-        sjson_transcripts = contentstore().find(content_location)
-        log.debug("Downloading subs for %s id", subs_id)
-        str_subs = generate_srt_from_sjson(json.loads(sjson_transcripts.data), speed=1.0)
-        if not str_subs:
-            log.debug('generate_srt_from_sjson produces no subtitles')
-            raise Http404
-        response = HttpResponse(str_subs, content_type='application/x-subrip')
-        response['Content-Disposition'] = 'attachment; filename="{0}.srt"'.format(subs_id)
-        return response
+        if not subs_id:
+            raise NotFoundError
+
+        filename = subs_id
+        content_location = StaticContent.compute_location(
+            item.location.course_key,
+            'subs_{filename}.srt.sjson'.format(filename=filename),
+        )
+        sjson_transcript = contentstore().find(content_location).data
     except NotFoundError:
-        log.debug("Can't find content in storage for %s subs", subs_id)
+        # Try searching in VAL for the transcript as a last resort
+        transcript = get_video_transcript_content(
+            course_id=item.location.course_key,
+            language_code=u'en',
+            edx_video_id=item.edx_video_id,
+            youtube_id_1_0=item.youtube_id_1_0,
+            html5_sources=item.html5_sources,
+        )
+        if not transcript:
+            raise Http404
+
+        filename = transcript['file_name'].split('.')[0].encode('utf8')
```

Comment:
This will not work if filename contains `.` characters.
```bash
>>> 'a.b.c.srt'.split('.')[0]
'a'
```
We can use os module to get filename
```bash
>>> import os
>>> os.path.splitext(os.path.basename('a.b.c.srt'))[0]
'a.b.c'
```